### ISSUE TITLE
[Module:Cache] createEntry failed, may due to no permission to write file or file is updating.

### ISSUE THREAD

**Issue by franciscopaniskaseker** [litespeedtech/openlitespeed#252]
I did not find here some answer about final solution about this.

**OS:** Centos 7.9.2009
**Openlitespeed:** openlitespeed-1.7.11-1.el7.x86_64

**Error (error.log):**
```bash
[Module:Cache] createEntry failed, may due to no permission to write file or file is updating.
```

**Directories permissions:**
```bash
[root@mail ~]# ls -l /usr/local/lsws/ | egrep -i cache
drwxr-xr-x.  3 nobody nobody  4096 Jun  7 16:20 cachedata
[root@mail ~]# 

[root@mail ~]# ls -l /usr/local/lsws/cachedata/
total 4
drwxrwx---. 18 nobody nobody 4096 Out 12  2020 priv
[root@mail ~]# ls -l /usr/local/lsws/cachedata/priv/
total 64
drwxrwx---. 18 nobody nobody 4096 Nov  3  2020 0
drwxrwx---. 18 nobody nobody 4096 Nov  2  2020 1
drwxrwx---. 18 nobody nobody 4096 Nov  3  2020 2
drwxrwx---. 18 nobody nobody 4096 Out 27  2020 3
drwxrwx---. 18 nobody nobody 4096 Out 23  2020 4
drwxrwx---. 18 nobody nobody 4096 Nov  2  2020 5
drwxrwx---. 18 nobody nobody 4096 Nov 13  2020 6
drwxrwx---. 18 nobody nobody 4096 Nov 21  2020 7
drwxrwx---. 18 nobody nobody 4096 Nov 20  2020 8
drwxrwx---. 18 nobody nobody 4096 Out 26  2020 9
drwxrwx---. 18 nobody nobody 4096 Nov  2  2020 a
drwxrwx---. 18 nobody nobody 4096 Nov 20  2020 b
drwxrwx---. 18 nobody nobody 4096 Nov  4  2020 c
drwxrwx---. 18 nobody nobody 4096 Out 17  2020 d
drwxrwx---. 18 nobody nobody 4096 Nov 19  2020 e
drwxrwx---. 18 nobody nobody 4096 Out 28  2020 f
[root@mail ~]# 
```

**Comment by PierreSp** [comment]
In case you are using VHosts:

I had the same issue and was able to solve it by setting the permission to the vhosts caching directory (priv/ .cacheman.lock and .cacheman.shm)

You can also try to set the owner to the user of lsws (or the extuser if you are using an external app / vhost. In general, I wouldn't use user nobody for production systems

**Comment by litespeedtech** [comment]
> [Module:Cache] createEntry failed, may due to no permission to write file or file is updating.
I think there is no permission problem. 
Likely because `file is updating` case, this has been fixed in 1.7.12, log the two cases separately. 
No worry.

**Comment by litespeedtech** [comment]
Release 1.7.12 should have addressed this.
